python/revng/cli/_commands/ptml/text.py

```diff
@@ -0,0 +1,120 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import re
+import sys
+from io import TextIOWrapper
+from typing import Dict
+from xml.dom import Node
+from xml.dom.minidom import Document, parseString
+
+from .common import handle_file, log, normalize_filter_extract
+
+COLOR_CONVERSION = {
+    "asm.label": "bright_red",
+    "asm.label-indicator": "bright_white",
+    "asm.comment-indicator": "white",
+    "asm.mnemonic": "dodger_blue1",
+    "asm.mnemonic-prefix": "bright_green",
+    "asm.mnemonic-suffix": "bright_green",
+    "asm.immediate-value": "dark_slate_gray2",
+    "asm.memory-operand": "dodger_blue1",
+    "asm.register": "orange1",
+    "c.function": "bright_red",
+    "c.type": "sea_green1",
+    "c.operator": "dodger_blue1",
+    "c.comparison": "dodger_blue1",
+    "c.function_parameter": "orange1",
+    "c.variable": "orange1",
+    "c.field": "orange1",
+    "c.constant": "dark_slate_gray2",
+    "c.string_literal": "medium_purple1",
+    "c.keyword": "dodger_blue1",
+    "c.directive": "dodger_blue1",
+}
+
+
+class PlainConsole:
+    def __init__(self, file: TextIOWrapper):
+        self.file = file
+
+    def print(self, text: str, *, end: str = "\n", **kwargs):  # noqa: A003
+        self.file.write(f"{text}{end}")
+
+    def __del__(self):
+        self.file.flush()
+
+
+def parse_ptml_plain(content: str, console):
+    dom = parseString(content)
+    _parse_ptml_node(dom, console, "", {})
+
+
+def parse_ptml_yaml(content: Dict[str, str], console):
+    for key, value in content.items():
+        console.print(f"{key}:", style="yellow1", end="")
+        console.print(" |-")
```

Comment:
Are you sure that using `|-` you can dump whatever you want and things works just by prefixing with `  `?
I'd feel safer if we'd just modify the document in place and use `yaml.dump`.

Reply:
There are 4 main issues:
1. By using yaml.dump the keys can't be colored as easiliy
2. Currently pyyaml needs a [slight hack](https://github.com/yaml/pyyaml/issues/121#issuecomment-1018117110) to make flow scalars work
3. `rich` works mainly on the terminal, to make it work with yaml I'd have to have a fake file-like object where rich prints in it and then repackage it in a dictionary, which is inefficient and would bloat the codebase
4. If the situation was more generic (i.e. we don't know ahead of time the structure of the output) I'd 100% agree with using `yaml.dump`, however the structure of the output is fixed, so there (as far as I see) no situations where the output would get broken

Reply:
OK, please add a comment about these reasons.

Reply:
Added